File: java/org.financer.client/src/main/java/org/financer/client/domain/model/user/Token.java
package org.financer.client.domain.model.user;

import org.financer.shared.domain.model.value.objects.ExpireDate;
import org.financer.shared.domain.model.value.objects.IPAddress;
import org.financer.shared.domain.model.value.objects.OperatingSystem;
import org.financer.shared.domain.model.value.objects.TokenString;

import java.util.Objects;

public class Token {
    private static final long serialVersionUID = 8834445127500149942L;

    private long id;
    private User user;
    private TokenString token;
    private ExpireDate expireDate;
    private IPAddress ipAddress;
    private OperatingSystem operatingSystem;

    /*
     * Getter and Setter
     */

    public long getId() {
        return id;
    }

    public Token setId(long id) {
        this.id = id;
        return this;
    }

    public User getUser() {
        return user;
    }

    public Token setUser(User user) {
        this.user = user;
        return this;
    }

    public TokenString getToken() {
        return token;
    }

    public Token setToken(TokenString token) {
        this.token = token;
        return this;
    }

    public ExpireDate getExpireDate() {
        return expireDate;
    }

    public Token setExpireDate(ExpireDate expireDate) {
        this.expireDate = expireDate;
        return this;
    }

    public IPAddress getIpAddress() {
        return ipAddress;
    }

    public Token setIpAddress(IPAddress ipAddress) {
        this.ipAddress = ipAddress;
        return this;
    }

    public OperatingSystem getOperatingSystem() {
        return operatingSystem;
    }

    public Token setOperatingSystem(OperatingSystem operatingSystem) {
        this.operatingSystem = operatingSystem;
        return this;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Token that = (Token) o;
        if (this.id >= 0) {
            return id == that.id;
        } else {
            return token.equals(that.token);
        }
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }
}

File: java/org.financer.client/src/main/kotlin/org/financer/client/domain/model/user/Token.kt
package org.financer.client.domain.model.user

import org.financer.shared.domain.model.value.objects.ExpireDate
import org.financer.shared.domain.model.value.objects.IPAddress
import org.financer.shared.domain.model.value.objects.OperatingSystem
import org.financer.shared.domain.model.value.objects.TokenString
import java.util.*

class Token {
    /*
     * Getter and Setter
     */ var id: Long = 0
        private set
    var user: User? = null
        private set
    var token: TokenString? = null
        private set
    var expireDate: ExpireDate? = null
        private set
    var ipAddress: IPAddress? = null
        private set
    var operatingSystem: OperatingSystem? = null
        private set

    fun setId(id: Long): Token {
        this.id = id
        return this
    }

    fun setUser(user: User?): Token {
        this.user = user
        return this
    }

    fun setToken(token: TokenString?): Token {
        this.token = token
        return this
    }

    fun setExpireDate(expireDate: ExpireDate?): Token {
        this.expireDate = expireDate
        return this
    }

    fun setIpAddress(ipAddress: IPAddress?): Token {
        this.ipAddress = ipAddress
        return this
    }

    fun setOperatingSystem(operatingSystem: OperatingSystem?): Token {
        this.operatingSystem = operatingSystem
        return this
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val that = o as Token
        return if (id >= 0) {
            id == that.id
        } else {
            token == that.token
        }
    }

    override fun hashCode(): Int {
        return Objects.hash(id)
    }

    companion object {
        private const val serialVersionUID = 8834445127500149942L
    }
}